### DISCUSSION TITLE
[error] [http client] internal response: 500 Can't connect to x.x.x.x:443

### DISCUSSION THREAD

**Discussion by mati294** [glpi-project/glpi-agent#312] [Q&A]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

Yes, I know

### Describe the bug

The agent cannot connect to the server via https.
```
[http client] internal response: 500 Can't connect to x.x.x.x:443 A connection could not be established because the target machine actively refused it., IO::Socket::IP configuration failed.
```
This happens with and without Disable SSL check.
Fingerprint of SSL certificate to thrust is not generated in the logfile.

The glpi server is on a linux virtual machine and an exception is added to the firewall for port 443.
Unfortunately, this happens with the firewall completely disabled

agent.cfg

```
# CA certificates directory
ca-cert-dir =
# CA certificates file
ca-cert-file =
# do not check server SSL certificate
no-ssl-check = 0
# connection timeout, in seconds
timeout = 180
```
```
[Mon Jan  9 10:37:05 2023][info] GLPI Agent exiting (7852)
[Mon Jan  9 10:37:09 2023][info] GLPI Agent service starting
[Mon Jan  9 10:37:09 2023][info] [http server] HTTPD service started on port 62354
[Mon Jan  9 10:37:10 2023][info] GLPI Agent memory usage: WSS=3788800 PFU=105689088
[Mon Jan  9 10:37:10 2023][info] target server0: next run: Mon Jan  9 10:37:09 2023 - https://192.168.32.10/glpi/front/inventory.php
[Mon Jan  9 10:37:10 2023][info] target server0: server https://192.168.32.10/glpi/front/inventory.php
[Mon Jan  9 10:37:10 2023][info] sending prolog request to server0
[Mon Jan  9 10:37:12 2023][error] [http client] internal response: 500 Can't connect to 192.168.32.10:443 (Nie można nawiązać połączenia, ponieważ komputer docelowy aktywnie go odmawia.), IO::Socket::IP configuration failed
[Mon Jan  9 10:37:12 2023][error] No answer from server at https://192.168.32.10/glpi/front/inventory.php
[Mon Jan  9 10:37:12 2023][info] target server0: next run: Mon Jan  9 10:38:12 2023 - https://192.168.32.10/glpi/front/inventory.php
```

### To reproduce

I just installed everything from the instructions on YT and on xamp it worked, or at least I think so...

### Expected behavior

Maybe I don't understand something and SSL works "automatically" that's why it is not generated in the log file but then https should work? Been working on it for a few weeks now.
Please help

### Operating system

Windows

### GLPI Agent version

1.4

### GLPI version

Other (See additional context below)

### GLPIInventory plugin or FusionInventory for GLPI plugin version

Not applicable

**Comment by g-bougard** [comment]
Hi @mati294 

I don't see any "bug" there so I moved your request to the discussion section.

> This happens with and without Disable SSL check.

Have you even tried without SSL, so with `http://192.168.32.10/glpi/front/inventory.php` ? Of course, the server must accept such url, at least a few time for the test.

You're speaking about linux agent but select "Windows" as OS. Do you have the same error with an agent running on linux and an agent running on windows ?
But you're talking about Xamp, so maybe, you selected "Windows" just because you installed GLPI on windows.
Anyway, "linux" is not very accurate. Can you tell on which OS you really had that error, telling which linux vendor & version you're using ?

And don't forget to tell how you installed the agent. Speaking about instructions you followed from YT we are not aware won't help us to help you.

**Reply by mati294** [reply]
The server is rejecting https connections and I don't know how to change it.

For http it works but requires:
1 Install the agent with the disable ssl check option.
2 After connecting to the disable ssl check server.

 Point 2 is optional but I would like it to be a secure connection.

The server is set up on Ubuntu 22.04.1 LTS
The agent is installed on Windows 10,
same problem on multiple versions.

When the server was tested on xamp, the ssl fingerprint appeared in the agent's log.

**Reply by g-bougard** [reply]
When your GLPI server has SSL enabled, can you share the output of the following command run from the computer where the agent is installed in an administrator console and from the installation folder ?
```
glpi-agent --config=none --debug --debug --logger=stderr --server=https://192.168.32.10/glpi/front/inventory.php --tasks=collect --no-ssl-check
```
The purpose of this command is to skip your config and debug at level 2 a GLPI server contact via SSL without any task to run (as you probably didn't setup any collect task for your agent).

If you have the exactly same error, you just probably have a server setup issue. In that case, are you even able to access your GLPI via the SSL url ?

**Reply by mati294** [reply]
The error is the same.
I can't connect via https either.
How to activate ssl on the server?

I did not see such an option in the server options and on the Internet I only found information about connecting via ssl to the sql database.

```
C:\Program Files\GLPI-Agent>glpi-agent --config=none --debug --debug --logger=stderr --server=https://192.168.32.10/glpi/front/inventory.php --tasks=collect --no-ssl-check
[debug] Logger backend Stderr initialized
[debug] GLPI Agent (1.4)
[debug] Configuration directory: C:/Program Files/GLPI-Agent/etc
[debug] Data directory: C:/Program Files/GLPI-Agent/share
[debug] Storage directory: C:\Program Files\GLPI-Agent\var
[debug] Lib directory: C:/Program Files/GLPI-Agent/perl/agent
[debug] [target server0] Next server contact planned for Tue Jan 10 08:26:59 2023
[debug2] getAvailableTasks() : add of task Collect version 2.8
[debug2] getAvailableTasks() : add of task Deploy version 2.10
[debug2] getAvailableTasks() : add of task Inventory version 1.12
[debug2] getAvailableTasks() : add of task RemoteInventory version 1.0
[debug2] Preparing execution plan
[debug] Available tasks:
[debug] - Collect: 2.8
[debug] - Deploy: 2.10
[debug] - Inventory: 1.12
[debug] - RemoteInventory: 1.0
[debug] target server0: server https://192.168.32.10/glpi/front/inventory.php
[debug] Planned tasks for server0: Collect
[debug] Provided by Teclib Edition
[debug] Installer built on Fri Jul  1 09:35:20 2022 UTC
[debug] Built with Strawberry Perl 5.36.0
[debug] Built on github actions windows image for glpi-project/glpi-agent repository
[debug] Running in foreground mode
[info] target server0: server https://192.168.32.10/glpi/front/inventory.php
[debug] [http client] Using Compress::Zlib for compression
[info] sending prolog request to server0
[debug2] [http client] sending message:
<?xml version="1.0" encoding="UTF-8" ?>
<REQUEST>
  <DEVICEID>L6-2023-01-09-08-13-52</DEVICEID>
  <QUERY>PROLOG</QUERY>
  <TOKEN>12345678</TOKEN>
</REQUEST>
[error] [http client] internal response: 500 Can't connect to 192.168.32.10:443 (Nie mo┐na nawi╣zaŠ po│╣czenia, poniewa┐ komputer docelowy aktywnie go odmawia.), IO::Socket::IP configuration failed
[error] No answer from server at https://192.168.32.10/glpi/front/inventory.php
```

**Reply by g-bougard** [reply]
Hi @mati294 
sorry, this is not the right place to request help to setup a SSL HTTP server. There's a lot of resource on internet to solve such problem, so take your time and do more deep search.
Here we would only like to focus on glpi-agent real problems.
Anyway maybe somebody else here have time to guide you.
